### PR TITLE
[CWS] add load_module and unload_module events

### PR DESCRIPTION
**Pull request by Gui774ume**
### What does this PR do?

This PR introduces 2 new kernel events:
- `load_module` can be used to track when a new kernel module is loaded
- `unload_module` can be used to track when a kernel module is deleted from the kernel

The following type of rules can now be written:
- `load_module.file.path not in [ my_list_of_trusted_modules ]`
- `load_module.name =~ "*"` to monitor all kernel module loading events
- `load_module.name =~ "*" && process.file.name not in [ my_list_of_trusted_module_loaders ]`
- `unload_module.name =~ "*"` to monitor all kernel module deletion events
- `load_module.loaded_from_memory == true` to trigger an alert whenever a kernel module is loaded from memory

### Motivation

Introduce a new event type to write new CWS rules.

### Describe how to test/QA your changes

Functional tests were added to test the new event types.

### Reviewer's Checklist

- [x] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [x] The appropriate `team/..` label has been applied, if known.
- [x] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [x] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [x] Changed code has automated tests for its functionality.
- [x] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [x] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [x] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [x] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

**Review by cswatt — APPROVED** [review]
docs look good

File: docs/cloud-workload-security/agent_expressions.md [hunk 1]
```diff
@@ -36,7 +36,9 @@ Triggers are events that correspond to types of activity seen by the system. The
 | `capset` | Process | A process changed its capacity set | 7.27 |
 | `chmod` | File | A file’s permissions were changed | 7.27 |
 | `chown` | File | A file’s owner was changed | 7.27 |
+| `delete_module` | Kernel | A kernel module was deleted | 7.35 |
```
Comment (lebauce) [inline docs/cloud-workload-security/agent_expressions.md:39]: `unload_module` | Kernel | A kernel module was unloaded

File: docs/cloud-workload-security/agent_expressions.md [hunk 2]
```diff
@@ -305,6 +307,15 @@ A file’s owner was changed
 | `chown.file.user` | string | User of the file's owner |
 | `chown.retval` | int | Return value of the syscall |
 
+### Event `delete_module`
```
Comment (lebauce) [inline docs/cloud-workload-security/agent_expressions.md:310]: s/delete_module/unload_module/

File: docs/cloud-workload-security/agent_expressions.md [hunk 3]
```diff
@@ -305,6 +307,15 @@ A file’s owner was changed
...
+
+A kernel module was deleted
```
Comment (lebauce) [inline docs/cloud-workload-security/agent_expressions.md:312]: s/deleted/unloaded

File: docs/cloud-workload-security/agent_expressions.md (hunk 1 above)
Comment (paulcacheux) [reply docs/cloud-workload-security/agent_expressions.md:39]: I think the doc just needs to be regenerated

File: pkg/security/secl/model/category.go [hunk 4]
```diff
@@ -34,7 +34,7 @@ func GetEventTypeCategory(eventType eval.EventType) EventCategory {
 	switch eventType {
 	case "exec":
 		return ProcessCategory
-	case "bpf", "selinux", "mmap", "mprotect", "ptrace":
+	case "bpf", "selinux", "mmap", "mprotect", "ptrace", "init_module", "delete_module":
```
Comment (lebauce) [inline pkg/security/secl/model/category.go:37]: s/init_module/load_module/
s/delete_module/unload_module/

File: pkg/security/tests/kernel_module_test.go [hunk 5]
```diff
@@ -0,0 +1,262 @@
...
+		},
+	}
+
+	test, err := newTestModule(t, nil, ruleDefs, testOpts{})
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer test.Close()
+
+	t.Run("init_module", func(t *testing.T) {
```
Comment (lebauce) [inline pkg/security/tests/kernel_module_test.go:145]: s/init_module/load_module/

File: pkg/security/tests/kernel_module_test.go [hunk 6]
```diff
@@ -0,0 +1,262 @@
...
+		},
+	}
+
+	test, err := newTestModule(t, nil, ruleDefs, testOpts{})
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer test.Close()
+
+	t.Run("delete_module", func(t *testing.T) {
```
[the remark this answers is not in the document]
Comment (Gui774ume) [reply pkg/security/tests/kernel_module_test.go:234]: not this one: this is the syscall name

File: pkg/security/tests/kernel_module_test.go (hunk 5 above)
Comment (Gui774ume) [reply pkg/security/tests/kernel_module_test.go:145]: syscall name again :)

**Review by lebauce — APPROVED** [review]

**Review by paulcacheux — APPROVED** [review]

**Pull request merged**